### DISCUSSION TITLE
Providing the Subtotal for Sale Items Excluding Line Items

### DISCUSSION THREAD

**Discussion by bradenfallon** [inventree/InvenTree#7594] [Q&A]
Hello, 

I am wondering if there is a pre-existing variable that contains the total for the sale items of a Sales Order but not any of the additional line items.

Below I have a segment of code from my Sales Order Template file. It generates a table with all the sale items and adds a subtotal row underneath. However, I don't know how to calculate only the sale items total excluding any line items. I know that the `order.total_price` variable contains the entire order total (including line items).

If there is not a variable that exists with this exact data, is there a way to code this in the Sale Order Report Template?

```
<table class='table table-striped table-condensed'>

    <thead>
        <tr>
            <th>{% trans "Item" %}</th>
            <th>{% trans "Reference" %}</th>
            <th>{% trans "Quantity" %}</th>
            <th>{% trans "Unit Price" %}</th>
            <th>{% trans "Total Price" %}</th>
        </tr>
    </thead>

    <body>

        {% for line in lines.all %}
        <tr>
            <td>
                <div class='thumb-container'>
                    <img src='{% part_image line.part height=240 %}' alt='' class='part-thumb'>
                    <span class='part-text'>
                        {{ line.part.name }}
                    </span>
                </div>
            </td>
            <td>{{ line.part.IPN }}</td>
            <td>{% decimal line.quantity %}</td>
            <td>{% render_currency line.price %}</td>
            <td>{% render_currency line.total_line_price %}</td>
        </tr>
        {% endfor %}

        <tr>
            <td colspan="3"></td>
            <th>{% trans "Subtotal" %}</th>
            <td>{% render_currency subtotal_variable currency=order.currency %}</td>
        </tr>
```

**Comment by SchrodingersGat** [comment]
@bradenfallon this does not currently exist as a pre-calculated variable - although it would be a nice addition and potentially not too difficult to add to the available context data in the report. I would suggest making a new issue against this. Perhaps you would be willing to put some sponsorship towards development of this feature?

**Reply by bradenfallon** [reply]
I will submit this as a a feature request. I have been looking through the code and have a little grasp on how everything is communicating. I think I have also isolated where the relevant data is calculated. I might be able to aid in coding this our at least provide a scaffold.
